File: backend/Tickets.Application/DTOs/Auth/AuthResponse.cs
using System.Collections.Generic;
using Tickets.Application.DTOs.Usuarios;

namespace Tickets.Application.DTOs.Auth
{
    /// <summary>
    /// DTO de respuesta de autenticación
    /// </summary>
    public class AuthResponse
    {
        /// <summary>
        /// Token de acceso JWT
        /// </summary>
        public string AccessToken { get; set; } = string.Empty;

        /// <summary>
        /// Token de refresco
        /// </summary>
        public string RefreshToken { get; set; } = string.Empty;

        /// <summary>
        /// Fecha de expiración del access token
        /// </summary>
        public DateTime ExpiresAt { get; set; }

        /// <summary>
        /// Información del usuario
        /// </summary>
        public UsuarioDto Usuario { get; set; } = null!;

        /// <summary>
        /// Roles del usuario
        /// </summary>
        public List<string> Roles { get; set; } = new();

        /// <summary>
        /// Permisos del usuario
        /// </summary>
        public List<string> Permisos { get; set; } = new();
    }
}

File: frontend/lib/data/models/auth/auth_response.dart
import '../user/user_model.dart';

class AuthResponse {
  final String accessToken;
  final String refreshToken;
  final DateTime expiresAt;
  final UserModel usuario;
  final List<String> roles;
  final List<String> permisos;

  AuthResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.expiresAt,
    required this.usuario,
    required this.roles,
    required this.permisos,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      accessToken: json['accessToken'] ?? '',
      refreshToken: json['refreshToken'] ?? '',
      expiresAt: DateTime.parse(json['expiresAt']),
      usuario: UserModel.fromJson(json['usuario']),
      roles: List<String>.from(json['roles'] ?? []),
      permisos: List<String>.from(json['permisos'] ?? []),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'accessToken': accessToken,
      'refreshToken': refreshToken,
      'expiresAt': expiresAt.toIso8601String(),
      'usuario': usuario.toJson(),
      'roles': roles,
      'permisos': permisos,
    };
  }
}
